ClickHouse/ClickHouse
# Remove unbundled support

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Remove unbundled support

Detailed description / Documentation draft:
Recently I found one more issues with unbundled builds, and instead of fixing it in particular I decided to do a global cleanup.
Note, that right now (since https://github.com/ClickHouse/ClickHouse/pull/30599/commits/d604cf55734d5aa2a2abb98b1cff04763ea414b8), to use UNBUNDLED build you need to manually set some variables, in other words there is no more one common switch for this (so it is unlikely that someone uses it, even I don't already)

Implementation:
- each library is in a separate commit (more or less)
- each commit may include some refactoring, since there was some issues while I was looking at this, one of common mistakes are:
  - not using internal bundled library (zlib)
  - use of non-existing variables (`X_LIBRARIES` instead of `X_LIBRARY`)
  - remove quirks for library types (statis/shared)
  - use `ENABLE_LIBRARIES` as default for contrib
  - move `-DUSE_XXX` to config file (`configure_file`)
- original library name hidden with a prefix (to avoid accidentally using it)
- alias that starts with ch_contrib:: is used as a public interface (that should include all include directories and compile definitions)
- add alias libraries even for header-only libraries
- **add a style check to avoid using `find_path`/`find_libraries` (as a marker for non-bundled/system-wide libraries)**

Checklist:
- [x] cleanup all `cmake/find/*.cmake`
- [x] cleanup docs
- [x] cleanup `USE_INTERNAL_`
- [x] introduce helper to include folder and check that submodule cloned

Fixes: #30598 (cc @alexey-milovidov )
Fixes: #9226
